### ISSUE TITLE
mocha & mocha sidebar does not find any test // windows typescript

### ISSUE THREAD

**Issue by mattmeye** [maty21/mocha-sidebar#145]
### Description

i think i installed mocha sidebar and all prerequisites, mocha runs without an error, but does not find any test file.

#### Steps to Reproduce

open vs code project (with test under /test/example.test.ts)

#### Expected Results
example is shown in the sidebar

#### Actual Results
sidebar is empty

#### Settings file & mocha verbose log output

```
____________________________________________________________________________
trying to searching for tests using these settings: 

    mocha path: mocha
    test files location: test/**/*.ts
    files to ignore: **/.git/**/*,**/node_modules/**/*
    environmets: {}
    requires: ["ts-node/register"]
    options:  {"compilers":{"ts":"ts-node/register"}}


if you find anything wrong please change those default settings
____________________________________________________________________________

test runs with those args:

  files: [],
  options: {"compilers":{"ts":"ts-node/register"}},
  grep:null,
  requires: ["ts-node/register"],
  rootpath: "c:\\dev\\p4-serverless"
  
Applying Mocha options:
  {
    "compilers": {
      "ts": "ts-node/register"
    }
  }



Test file(s):





  0 passing (1ms)


###end

------------------------------------

finish failure amount:0
------------------------------------
```

### Versions

- VScode: v1.87.2
- Mocha SideBar: v0.20.18 
- Node Version: v10.8.0
- "ts-node": "^7.0.0",

### OS version
- [x] Windows

**Comment by mattmeye** [comment]
i didn't set the rootpath manually in my project, even the log tells anything else.

**Comment by mattmeye** [comment]
``` 
mocha -r ts-node/register **/*.test.ts

C:\Users\xxx\Desktop\mocha-sidebar-test\test\example.test.ts:4
describe('titleA', () => {
^
TypeError: mocha_1.describe is not a function
````

**Comment by mattmeye** [comment]
I fixed this error in the repo and I'm able to run more mocha manually, but mocha sidebar still doesn't find any tests

**Comment by weagle08** [comment]
I'm having the same issue. No tests are appearing except our code is in TS but all the tests are written in JS.

**Comment by GeneticSYS** [comment]
@mattmeye we faced the same issue and solved it with these user settings within VS Code:

```json
{
"mocha.requires": [
    "ts-node/register"
],
"mocha.files.glob": "./test/**/*.test.ts"
}
```

**Comment by maty21** [comment]
> i didn't set the rootpath manually in my project, even the log tells anything else.
> i created my demo there : https://gitlab.com/itsolutionsmeyer/mocha-sidebar-test

test it on my pc and it's working correctly (but in ubuntu )
i'll try to find time for testing it on windows

**Comment by mattmeye** [comment]
hey,

i have news: when i restart vsc fully i see the tests in the mocha sidebar, but if i reload the window (it do it sometimes) the tests are not visible. 
@maty21 maybe you can try to test this on ubuntu too

**Comment by 3DEsprit** [comment]
I had this exact issue on OSX when setting up for TS. To troubleshoot, I started by removing the options and subDir settings, but afterwards the ability to access the commands were lost completely (errors on all command palette commands). 

From this point, I did the following to get it working:

- removed all mocha settings
- uninstalled extension and reloaded 
- reinstalled extension
- renamed spec title removing the word *testing* (this showed result changes in previous install)

Setup:

- updated glob to relative test folder ./spec/app/**/*.spec.ts   
- set requires with single require, "ts-node/register" 
- used command palette command "Run All Tests"

**To help reproduce the issue**: 
------------------------------------------------------------
**Install - First install from start to finish**: 
- I temporarily set the **explorer view** 
- followed the first Q&A:
  - set **file glob** 
  - set **mocha path** using **./node_modules/.bin/mocha** (_did not read description correctly_)
  - set the **subdirectory** to _spec_ folder
  - added **requires** with _ts-node/register_ and my _test-setup.js_ file (_for non-sidebar tests_)
  - set **options** _ui, recursive, sort, timeout, retries_ 

NOTE: In the **User Settings** there was an empty option for **mocha.reporters**, which is not found with second install, but no other mocha extension has been installed. 

**Usage**
- When running or refreshing, test view was empty
- Output showed the sidebar attempting to run  
   - files array was always empty
   - also showed empty options array
   - no other output was shown after the settings
   - rootpath showed correct path
   - required showed correctly

**Troubleshooting Updates**
- removed options, leaving setting empty
- removed extra require
- removed subdirectory setting
- updated mocha path to default (commands were lost after this)

I hope that helps give some context as to what is happening!

If I could suggest something, it would be helpful to see the other paths in the output, also I was surprised there was not an error for the mocha command after being set incorrectly. Love the extension, ⭐️!

**Comment by maty21** [comment]
Hi I'm checking it send code sample please

**Comment by mattmeye** [comment]
@maty21 i think the problem of 3DEsprit is a little bit diffrent. you can see check my "window reload problem" with my sample source above.

**Comment by 3DEsprit** [comment]
Yes, my apologies for that. I didn't want to add another issue when it seemed related.

**Comment by maty21** [comment]
Did you check possible solution on issue #157 ?
